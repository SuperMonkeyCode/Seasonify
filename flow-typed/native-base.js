/* @flow */
declare module 'native-base' {
  declare var exports: any;
}
