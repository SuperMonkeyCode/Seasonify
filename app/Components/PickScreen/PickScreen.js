/* @flow */
import { Container, Content, ListItem, Radio, Right, Text } from 'native-base';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  navigation: {
    goBack: Function,
    state: {
      params: {
        title: string,
        isMultiSelection?: boolean,
        onSelect: Function,
        // Can be either an object or an array of {id, name}
        collection: Array<Object> | Object,
      },
    },
  },
  /* connect */
  /* HOC */
  handleItemSelect: Function,
};

const enhance = compose(
  withHandlers({
    handleItemSelect: ({ navigation }: Props) => itemId => {
      const { isMultiSelection, onSelect } = navigation.state.params;
      onSelect(itemId);
      /* istanbul ignore else */
      if (!isMultiSelection) navigation.goBack();
    },
  }),
  pure,
);

function PickScreen({ navigation, handleItemSelect }: Props) {
  const { collection } = navigation.state.params;
  let items = [];
  if (typeof items === 'object' && !Array.isArray(collection)) {
    items = Object.keys(collection).map(id => ({
      id: collection[id].id,
      name: collection[id].name,
    }));
  } else {
    items = collection;
  }
  return (
    <Container>
      <Content>
        {items.map(item =>
          (<ListItem key={item.id} onPress={() => handleItemSelect(item.id)}>
            <Text>
              {item.name}
            </Text>
            <Right>
              <Radio selected={false} onPress={() => handleItemSelect(item.id)} />
            </Right>
          </ListItem>),
        )}
      </Content>
    </Container>
  );
}

export default enhance(PickScreen);
