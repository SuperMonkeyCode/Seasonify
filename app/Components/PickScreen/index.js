/* @flow */
import { Body, Button, Header, Icon, Left, Title } from 'native-base';
import React from 'react';

import PickScreen from './PickScreen';

// $FlowFixMe
PickScreen.navigationOptions = ({ navigation }) => {
  const { title, isMultiSelection } = navigation.state.params;

  return {
    header: (
      <Header>
        <Left>
          <Button transparent onPress={() => navigation.goBack()}>
            <Icon name={isMultiSelection ? 'checkmark' : 'arrow-back'} />
          </Button>
        </Left>
        <Body>
          <Title>
            {title}
          </Title>
        </Body>
      </Header>
    ),
  };
};

export default PickScreen;
