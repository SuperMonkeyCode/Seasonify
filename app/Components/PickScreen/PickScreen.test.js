import { Container, ListItem, Radio } from 'native-base';
import React from 'react';

import PickScreen from './PickScreen';

function setup(specificProps = {}) {
  const props = {
    navigation: {
      goBack: jest.fn(),
      setParams: jest.fn(),
      state: {
        params: {
          title: 'Select something',
          isMultiSelection: false,
          onSelect: jest.fn(),
          collection: [{ id: 'a', name: 'Item A' }, { id: 'b', name: 'Item B' }],
        },
      },
    },
    ...specificProps,
  };
  const component = shallowDive(<PickScreen {...props} />, Container);
  return {
    component,
    props,
    listItem: component.find(ListItem).first(),
    radio: component.find(Radio).first(),
  };
}

describe('Rendering when default state', () => {
  const { component } = setup();
  it('should match', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('Rendering with an object collection', () => {
  const { component } = setup({
    navigation: {
      goBack: jest.fn(),
      setParams: jest.fn(),
      state: {
        params: {
          title: 'Select something',
          isMultiSelection: false,
          onSelect: jest.fn(),
          collection: { a: { id: 'a', name: 'Item A' }, b: { id: 'b', name: 'Item B' } },
        },
      },
    },
  });

  it('should match', () => {
    expect(component).toMatchSnapshot();
  });
});

/* ========= Events & Functions ========= */

describe('Events & Functions', () => {
  const { props, listItem, radio } = setup();

  describe('when calling onPress on listItem or radio', () => {
    listItem.props().onPress();
    radio.props().onPress();

    it('calls ', () => {
      expect(props.navigation.state.params.onSelect).toBeCalledWith('a');
      expect(props.navigation.goBack).toBeCalled();
    });
  });
});
