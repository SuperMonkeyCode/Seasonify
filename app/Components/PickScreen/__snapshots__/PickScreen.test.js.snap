// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering when default state should match 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <Styled(ListItem)
      onPress={[Function]}
    >
      <Styled(Text)>
        Item A
      </Styled(Text)>
      <Styled(Right)>
        <Styled(Radio)
          onPress={[Function]}
          selected={false}
        />
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      onPress={[Function]}
    >
      <Styled(Text)>
        Item B
      </Styled(Text)>
      <Styled(Right)>
        <Styled(Radio)
          onPress={[Function]}
          selected={false}
        />
      </Styled(Right)>
    </Styled(ListItem)>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`Rendering with an object collection should match 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <Styled(ListItem)
      onPress={[Function]}
    >
      <Styled(Text)>
        Item A
      </Styled(Text)>
      <Styled(Right)>
        <Styled(Radio)
          onPress={[Function]}
          selected={false}
        />
      </Styled(Right)>
    </Styled(ListItem)>
    <Styled(ListItem)
      onPress={[Function]}
    >
      <Styled(Text)>
        Item B
      </Styled(Text)>
      <Styled(Right)>
        <Styled(Radio)
          onPress={[Function]}
          selected={false}
        />
      </Styled(Right)>
    </Styled(ListItem)>
  </Styled(Content)>
</Styled(Container)>
`;
